Stripe/PublicHeaders/STPSourceInfoViewController.h

```diff
@@ -0,0 +1,59 @@
+//
+//  STPSourceInfoViewController.h
+//  Stripe
+//
+//  Created by Ben Guo on 2/9/17.
+//  Copyright © 2017 Stripe, Inc. All rights reserved.
+//
+
+#import <Stripe/Stripe.h>
+#import "STPCoreTableViewController.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+@protocol STPSourceInfoViewControllerDelegate;
+@class STPSourceParams;
+
+/** You can use this view controller to collect information for payment sources that require additional information upon creation (e.g. iDEAL, which requires your customer's name and bank.) Note that this view controller renders a right bar button item that submits the form, so it must be shown inside a `UINavigationController`.
+ */
+@interface STPSourceInfoViewController : STPCoreTableViewController
+
+/**
+ *  Initializes a new `STPSourceInfoViewController` with the provided parameters. When the user submits the form, the view controller will return a copy of the provided `STPSourceParams` object to its delegate, updated with the information entered by the user.
+ *
+ *  @param sourceParams     The source parameters to collect additional info for. @see STPSourceParams
+  *  @param theme           The theme to use to inform the view controller's visual appearance. @see STPTheme
+ */
+- (nullable instancetype)initWithSourceParams:(STPSourceParams *)sourceParams
+                                        theme:(STPTheme *)theme;
+
+/**
+ *  The view controller's delegate. This must be set before showing the view controller in order for it to work properly. @see STPSourceInfoViewControllerDelegate
+ */
+@property(nonatomic, weak) id<STPSourceInfoViewControllerDelegate> delegate;
```

Comment:
Hm – I'd prefer to leave this as a delegate, just because it's the pattern we've been using elsewhere. I think if we decide to move to completion blocks we should do it across STPAddCardVC, STPAddSourceVC, and STPSourceInfoVC, since they all just have finished and cancelled callbacks.